script.myepisodecalendar/main.py

```diff
@@ -0,0 +1,428 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import re
+import sys
+import threading
+import xbmc, xbmcaddon
+
+__addon__         = xbmcaddon.Addon()
+__addonid__       = __addon__.getAddonInfo('id')
+__cwd__           = __addon__.getAddonInfo('path')
+__icon__          = __addon__.getAddonInfo("icon")
+__scriptname__    = __addon__.getAddonInfo('name')
+__version__       = __addon__.getAddonInfo('version')
+__language__      = __addon__.getLocalizedString
+__resource_path__ = os.path.join(__cwd__, 'resources', 'lib')
+__resource__      = xbmc.translatePath(__resource_path__).decode('utf-8')
+__datapath__      = os.path.join(xbmc.translatePath('special://masterprofile/addon_data/').decode('utf-8'), __addonid__)
+
+DB_CACHE_TVDB_IDS = 'cache.tvdb.ids.db'
+
+__nextAired__     = False
+
+if __addon__.getSetting('tvsna-enabled') and xbmc.getCondVisibility('System.HasAddon(%s)' % 'script.tv.show.next.aired') == 1:
+    __nextAired__     = xbmcaddon.Addon(id='script.tv.show.next.aired')
+    addon_path = __nextAired__.getAddonInfo('path')
+    sys.path.append (xbmc.translatePath( os.path.join(addon_path) ))
+    sys.path = [xbmc.translatePath( os.path.join(addon_path, 'resources', 'lib') )] + sys.path
+
+    from default import NextAired
+    import xbmcgui
+
+from resources.lib.myepisodecalendar import MyEpisodeCalendar
+
+class Monitor(xbmc.Monitor):
+    def __init__( self, *args, **kwargs ):
+        xbmc.Monitor.__init__( self )
+        self.action = kwargs['action']
+
+    def onSettingsChanged( self ):
+        log('#DEBUG# onSettingsChanged')
+        # remove shows added by plugin when integration is disabled
+        if __nextAired__ and __addon__.getSetting('tvsna-enabled') == 'false':
+            __nextAired__.setSetting("ExtraShows", re.sub(r"\D?mec\d+", '', __nextAired__.getSetting("ExtraShows")))
```

Comment:
Probable an oversight by the previous reviewer but normally we do not allow addons to manipulate settings of other addons.

Reply:
Ok, I see... So I'll have to remove the next aired functionality completely, that's a bummer.

Reply:
Can you explain what this tries to accomplish?

Reply:
The integration, when enabled,  adds all your shows from myepisodecalendar.com to the list of shows of the TV Show Next Aired addon. And this specific part of the code removes those added shows again after saving the settings and having disabled the integration.